install/kubernetes/helm/subcharts/mixer/templates/deployment.yaml

```diff
@@ -274,6 +274,15 @@ metadata:
     istio: mixer
 spec:
   replicas: {{ $.Values.replicaCount }}
+  selector:
```

Comment:
i don't understand why all this is necessary.  Would you mind expanding on rationale?

Cheers
-steve

Reply:
context: https://github.com/kubernetes/kubernetes/issues/26202

`extensions/v1beta` Deployments don't require the selector. Instead, these are defaulted server-side based on the pod template spec labels. When we changed the pod template spec in 1.1.0, the defaulted selector no longer matched the _new_ pod template spec. This results in a `selector` does not match template `labels` error.

`extensions/v1beta` is deprecated and we should switch to `apps/v1betaX` soon. Starting in `apps/v1beta2` Deployments, an explicit selector is required.

What's also interesting is that the selector becomes immutable with `apps/v1beta2` Deployments (see https://github.com/kubernetes/kubernetes/pull/50719). It looks like this may break workflows that encode release versions in labels, e.g. helm (see https://github.com/helm/charts/issues/7680).